common/evidence.py

```diff
@@ -103,87 +145,87 @@ def generate_map(self) -> None:
             .persist()
         )
 
-        # Generating a unique set of tissues in a pandas dataframe:
-        tissues = cell_df.select("tissue").distinct().toPandas()
-        logging.info(f"Found {len(tissues)} tissues.")
-
-        # Generating a unique set of cell lines in a pandas series:
-        # TODO: UBERON IDs should not be mapped via REST queries. This should be done via joining with owl data.
-        mapped_tissues = tissues.assign(
-            tissueId=lambda df: df.tissue.apply(self.lookup_uberon)
-        )
-        logging.info(
-            f"Found mapping for {len(mapped_tissues.loc[mapped_tissues.tissueId.notna()])} tissues."
-        )
-
-        # Converting to spark dataframe:
-        mapped_tissues_spark = self.spark.createDataFrame(mapped_tissues)
-
         # Joining with cell lines:
         return (
-            cell_df.join(mapped_tissues_spark, on="tissue", how="left")
+            cell_df
             # Generating the diseaseCellLines object:
             .select(
                 f.regexp_replace(f.col("name"), "-", "").alias("name"),
                 "id",
                 "biomarkerList",
-                f.struct(["tissue", "name", "id", "tissueId"]).alias("diseaseCellLine"),
+                f.struct(
+                    f.col("tissueName").alias("tissue"), "name", "id", "tissueId"
+                ).alias("diseaseCellLine"),
             )
         )
 
-    def get_mapping(self):
-        return self.cell_map
+    def read_uberon_mapping(self, cell_line_to_uberon_mapping: str) -> DataFrame:
+        """Read the cell line to UBERON mapping file into a Spark dataframe (http or local).
 
-    @staticmethod
-    def lookup_uberon(tissue_label: str) -> str:
-        """Mapping tissue labels to tissue identifiers (UBERON codes) via the OLS API."""
+        Returned schema:
+            root
+            |-- tissue: string (nullable = true)
+            |-- tissueId: string (nullable = true)
+            |-- tissueName: string (nullable = true)
 
-        url = f"https://www.ebi.ac.uk/ols/api/search?q={tissue_label.lower()}&queryFields=label&ontology=uberon&exact=true"
-        r = requests.get(url).json()
+        Args:
+            cell_line_to_uberon_mapping (str): Path to the cell line to UBERON mapping file.
 
-        if r["response"]["numFound"] == 0:
-            return None
-        else:
-            return r["response"]["docs"][0]["short_form"]
+        Returns:
+            DataFrame: Spark dataframe containing the cell line to UBERON mapping.
+        """
 
-    @staticmethod
-    @f.udf(
-        ArrayType(
-            StructType(
-                [
-                    StructField("name", StringType(), nullable=False),
-                    StructField("description", StringType(), nullable=False),
-                ]
+        # If the mapping file is a URL, download it and read it into a Spark dataframe.
+        if "http" in cell_line_to_uberon_mapping:
```

Comment:
I THINK using SparkFiles with a local filepath also worked, but I don't remember exactly. Did you test it didn't work before coding both scenarios?

Reply:
I have tried both remote and local files, both work.